plugins/extraction/extract_env.ml

```diff
@@ -246,6 +281,13 @@ and extract_mexpr_spec table venv env mp1 (me_struct_o,me_alg) = match me_alg wi
     let () = Visit.add_mp_all venv mp in
     MTident mp
   | MEwith(me',WithDef(idl,(c,ctx)))->
+      let () = match ctx with
+      | None -> ()
+      | Some auctx ->
+        (* TODO *)
+        if Array.is_empty (UVars.AbstractContext.names auctx).quals then ()
+        else user_err Pp.(str "with Definition clauses not allowed for sort polymorphic definitions")
```

Comment:
~~~suggestion
        else user_err Pp.(str "Extraction of \"with Definition\" clauses not supported for sort polymorphic definitions.")
~~~